util/logtools/encounter_tools.ts

```diff
@@ -286,3 +286,191 @@ export class EncounterCollector extends EncounterFinder {
+    if (!str)
+      return '';
+    if (!length)
+      return str;
+    while (str.length < length)
+      str = ' ' + str;
+    return str;
+  }
+
+  rightExtendStr(str: string | undefined, length: number | undefined): string {
+    if (!str)
+      return '';
+    if (!length)
+      return str;
+    while (str.length < length)
+      str += ' ';
+    return str;
+  }
+
+  generateFileName(fightOrZone: FightEncInfo | ZoneEncInfo): string {
+    const zoneName = fightOrZone.zoneName || 'Unknown_Zone';
+    const dateStr = this.dayFromDate(fightOrZone.startTime).replace(/-/g, '');
+    const timeStr = this.timeFromDate(fightOrZone.startTime).replace(/:/g, '');
+    const duration = this.durationFromDates(fightOrZone.startTime, fightOrZone.endTime);
+    let seal;
+    if ('sealName' in fightOrZone)
+      seal = fightOrZone['sealName'];
+    if (seal)
+      seal = '_' + this.toProperCase(seal).replace(/[^A-z0-9]/g, '');
+    else
+      seal = '';
+    let wipeStr = '';
+    if ('endType' in fightOrZone)
+      wipeStr = fightOrZone.endType === 'Wipe' ? '_wipe' : '';
+    return `${zoneName}${seal}_${dateStr}_${timeStr}_${duration}${wipeStr}.log`;
+  }
+
+// For an array of arrays, return an array where each value is the max length at that index
+// among all of the inner arrays, e.g. find the max length per field of an array of rows.
+  maxLengthPerIndex(outputRows: Array<Array<string>>): Array<number> {
+    const outputSizes = outputRows.map((row) => row.map((field) => field.length));
+    return outputSizes.reduce((max, row) => {
+      return max.map((val, idx) => {
+        const indexed = row[idx];
+        if (indexed !== undefined)
+          return Math.max(val, indexed);
+        return 0;
+     });
+    });
+  }
+
+  printCollectedZones(collector: EncounterCollector): void {
+    let idx = 1;
+    const outputRows = [];
+    for (const zone of collector.zones) {
+      const zoneName = zone.zoneName ?? 'Unknown_Zone';
+      const startDate = zone.startTime ? this.dayFromDate(zone.startTime) : 'Unknown_Date';
+      const startTime = zone.startTime ? this.timeFromDate(zone.startTime) : 'Unknown_Start';
+
+      outputRows.push([
+        idx.toString(),
+        startDate,
+        startTime,
+        this.durationFromDates(zone.startTime, zone.endTime),
+        zoneName,
+      ]);
+      idx++;
+    }
+
+    if (outputRows.length === 0)
+      return;
+
+    const lengths = this.maxLengthPerIndex(outputRows);
+
+    const dateIdx = 1;
+    let lastDate = null;
+    for (const row of outputRows) {
+      if (row[dateIdx] !== lastDate) {
+        lastDate = row[dateIdx];
+        console.log(lastDate);
+      }
+      const row0 = this.leftExtendStr(row[0], lengths[0]);
+      const row2 = this.leftExtendStr(row[2], lengths[2]);
+      const row3 = this.leftExtendStr(row[3], lengths[3]);
+      const row4 = this.rightExtendStr(row[4], lengths[4]);
+
+      console.log(`  ${row0})   ${row2}   ${row3}  ${row4}`);
```

Comment:
Here and for `printCollectedFights`, I'm definitely not proposing that we leave this forever, by any means, but it does *work*, and it's not *that* slow. (Also I don't want to actually have to think about the logic right now. >_>)